apps/bthome/settings.js

```diff
@@ -11,10 +11,24 @@
     require("Storage").writeJSON("bthome.json",settings)
   }
 
+  function getNewIdNumber(){
+    var id_start = 1;
+    var select = false;
+    while(!select){
+      select = true;
+      settings.buttons.forEach((button, idx) => {
+        if (id_start == button.n){
+          id_start++;
+          select = false;
+        }
+      });
+    }
```

Comment:
I would personally write this like
```
const id_start = [1, 2, 3].find(id => settings.buttons.every(button => id != button.n));
```

Reply:
This is not bad suggestion. It would limit the choices to first array options and I liked that it was "limitless". But in practice, you would have max 5-6 buttons ( if you have more that 10, then you have to scroll a lot and I don't think it is really practical) . I will use this code with array off 1-9. This is good one. Thanks!